core/templates/dev/head/services/AudioPlayerService.js

```diff
@@ -30,6 +33,18 @@ oppia.factory('AudioPlayerService', [
             var blobUrl = URL.createObjectURL(audioBlob);
             _currentTrack = ngAudio.load(blobUrl);
             _currentTrackFilename = filename; 
+
+            // Doesn't seem to be any way of detecing when native
+            // audio object has finished loading. Use a timeout
+            // to grab native audio.
+            // TODO(tjiang11): Find a better way to handle this.
+            $timeout(function() {
+              _currentTrack.audio.onended = function() {
```

Comment:
Can you elaborate a bit more (in comments) on the ideas behind the workaround here? I'm not sure I fully understand it.